llvm/lib/Target/RISCV/RISCVInstrInfo.h

```diff
@@ -78,6 +78,10 @@ class RISCVInstrInfo : public RISCVGenInstrInfo {
 
   bool isReallyTriviallyReMaterializable(const MachineInstr &MI) const override;
 
+  bool shouldBreakCriticalEdgeToSink(MachineInstr &MI) const override {
+    return MI.getOpcode() == RISCV::ADDI;
```

Comment:
Is this always an ADDI with X0 source?

Reply:
It could be.  I left it unconstrained thinking we should probably always be sinking the addi, but if you want, I can add the restriction and it should change any of the diffs.